### PR TITLE
[Refactor] Remove shape from MultiHeadAttention

### PR DESCRIPTION
**Pull request by langong347**
Summary:
- Remove `shape` parameter from `MultiHeadAttention` (following removal from `FullAttention`)
- Restore logic to create causal attention mask in a new utility `torchmultimodal.utils.attention.get_causal_attention_mask`

Test Plan:
```
(torchmm) langong-mbp:gpt_attention langong$ python -m pytest test/utils/test_attention_utils.py -vv -rP
================================================= test session starts ==================================================
platform darwin -- Python 3.8.13, pytest-7.1.2, pluggy-1.0.0 -- /Users/langong/local/miniconda3/envs/torchmm/bin/python
cachedir: .pytest_cache
rootdir: /Users/langong/gpt_attention, configfile: pyproject.toml
plugins: mock-3.8.2, cov-3.0.0
collected 1 item

test/utils/test_attention_utils.py::test_get_causal_attention_masks PASSED                                             [100%]

========== PASSES ========
========== 1 passed in 1.13s =======
```

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #176
* #175
* __->__ #166

Differential Revision: [D38035203](https://our.internmc.facebook.com/intern/diff/D38035203)

### CODE REVIEW

**Review by RdoubleA — APPROVED** [review]

**Comment by langong347** [comment]
@langong347 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38035203).

**Comment by langong347** [comment]
@langong347 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38035203).
